### PR TITLE
[RFC] API: nvim_exec: fix script scope

### PR DESCRIPTION
**Pull request by in42**
* Fix script scope variables

Based on #11507

### CODE REVIEW

File: test/functional/api/vim_spec.lua [hunk 1]
```diff
@@ -115,6 +115,15 @@ describe('API', function()
...
       eq('Hello', request('nvim_eval', 'g:x1'))
+
+      -- Script scope (s:)
+      eq('ahoy! script-scoped varrrrr', nvim('exec', [[
+          let s:pirate = 'script-scoped varrrrr'
+          func! s:avast_ye_hades(s) abort
+            return s.' '.s:pirate
+          endf
+          echo <sid>avast_ye_hades('ahoy!')
+        ]], true))
```
Comment (bfredl) [inline test/functional/api/vim_spec.lua:126]: maybe we also want a test for the scope remaining after the fact. I mean a global function returning another `s:` var. and a second `eq` using `nvim_eval` or something (which should not have the `s:` scope set, could even test that `s:pirate` _fails_ in that context).

Comment (in42) [reply test/functional/api/vim_spec.lua:126]: I have added 2 more tests. Do they take care of the cases?

File: test/functional/api/vim_spec.lua [hunk 2]
```diff
@@ -115,6 +115,29 @@ describe('API', function()
...
+      -- Script scope (s:)
+      eq('ahoy! script-scoped varrrrr', nvim('exec', [[
+          let s:pirate = 'script-scoped varrrrr'
+          func! s:avast_ye_hades(s) abort
+            return a:s.' '.s:pirate
+          endf
+          echo <sid>avast_ye_hades('ahoy!')
+        ]], true))
+
+      eq('script-scoped varrrrr', nvim('exec', [[
```
Comment (bfredl) [inline test/functional/api/vim_spec.lua:128]: This case is currently the same as last one. What I mean is you want to make it `func! Avast_ye_hades()`. and then call it in a second `nvim_eval` api request from the lua client.

Comment (in42) [reply test/functional/api/vim_spec.lua]: I have the added the test case.

**Comment by in42** [comment]
The implementation seem to be fine but some tests seem to be failing due to unrelated issues.

**Pull request closed without merging**